test/GitVersionParser.js

```diff
@@ -0,0 +1,79 @@
+import log from "npmlog";
+
+// file under test
+import GitVersionParser from "../src/GitVersionParser";
+
+// silence logs
+log.level = "silent";
+
+describe("GitVersionParser", () => {
+
+  describe("parseVersion - without prefix", () => {
+    let parser;
+
+    beforeEach(() => {
+      parser = new GitVersionParser("");
+    });
+
+    it("should work for semver version", () => {
+      expect(parser.parseVersion("0.0.2")).toEqual({
+        prefix: null,
+        version: "0.0.2"
+      });
+
+      expect(parser.parseVersion("~0.0.2")).toEqual({
+        prefix: null,
+        version: "~0.0.2"
+      });
+    });
+
+    it("should work for git url", () => {
+      expect(parser.parseVersion("github:user-foo/project-foo#v0.0.1")).toEqual({
+        prefix: "github:user-foo/project-foo#",
+        version: "v0.0.1"
+      });
+
+      expect(parser.parseVersion("git@github.com:user-foo/project-foo#0.0.5")).toEqual({
+        prefix: "git@github.com:user-foo/project-foo#",
+        version: "0.0.5"
+      });
+    });
+  });
+
+  describe("parseVersion - with prefix", () => {
```

Comment:
[minor]: I would rename `with prefix` into `with versionPrefix`. The title confused me with the `prefix` property returned by the parser.